docs/rfcs/rfc-040-relay-first-api.md

```diff
@@ -0,0 +1,1282 @@
+```
+
+The `*Connection` types will be a Relay compliant [Connection Type](https://relay.dev/graphql/connections.htm#sec-Connection-Types). With the following fields:
+
+```graphql
+type MoviesConnection {
+    pageInfo: PageInfo! # Required by relay
+    edges: [MovieEdge!]! # Required by relay
+    totalCount: Int!
+    aggregation: MoviesAggregation
+}
+```
+
+A simple query would look like:
+
+```graphql
+query MoviesTitles {
+    moviesConnection {
+        edges {
+            node {
+                title
+            }
+        }
+    }
+}
+```
+
+### Edges
+
+The `*Edge` types will be a Relay [Edge Type](https://relay.dev/graphql/connections.htm#sec-Edge-Types). These fields will be used to traverse the API and hold the relationship properties, if any:
+
+```graphql
+type MovieActorsEdge {
+    cursor: String! # Required by Relay
+    node: PersonNode! # Required by Relay
+    fields: ActedIn # Fields of the relationship
+}
+```
+
+There will be an Edge type per entity, as well as a different Edge type per relationship, relationship edges may contain the property `fields` that will hold the relationship properties.
+
+A query following an edge and its properties would look like:
+
+```graphql
+query MoviesWithActors {
+    moviesConnection {
+        edges {
+            node {
+                title
+                actors {
+                    edges {
+                        node {
+                            name
+                        }
+                        fields {
+                            year
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+In this example, we get the related Actor name, as well as the property `year` from the relationship.
+
+### Nodes
+
+The Entity (e.g. movies) properties for the projection will be defined in a `*Node` type:
+
+```graphql
+type MovieNode {
+    title: String!
+    released: Int
+    alternativeTitles: [String!]!
+    actors(
+        where: PersonConnectionWhere
+        first: Int
+        after: String
+        last: Int
+        before: String
+        directed: Boolean = true
+        sort: [MovieActorsConnectionSort!]
+    ): MovieActorsConnection!
+    director(where: MovieDirectorConnectionWhere, directed: Boolean = true): MovieDirectorConnection!
+}
+```
+
+Note that this type does **not** contain any operations in itself, it has the same properties as the user-defined entity, with the only difference being the result of the nested `actors` and `director` types, that support the top-level input and return a `*Connection`.
+
+### Filtering
+
+Filtering follows the same structure as projections, following nodes and edges. The operations `AND`, `OR` and `NOT` are available at any level. Comparison operators are present inside each field.
+
+```graphql
+query MatrixMoviesFrom1999 {
+    moviesConnection(
+        where: { edges: { node: { AND: [{ title: { contains: "Matrix" } }, { released: { eq: 1999 } }] } } }
```

Comment:
Could we consider making all operators uppercased? I find it a bit confusing to keep just `AND`, `OR` and `NOT` uppercased, and the other operators "blending" with user-defined properties

Reply:
Yeah I agree, it would be much clearer what is a field and what is an operator!

Reply:
Nice suggestion!

Reply:
TBH I find quite ugly uppercased operators

the reason for AND OR NOT being uppercased is mainly that they exist within the data objects and need to be differentiated, but it is an aesthetic decision so we can change it if most people agree is better to have all of them being uppercase

Reply:
I agree with both the opinion honestly.

Reply:
I'll add this to further discussion points